internal/shader/expr.go

```diff
@@ -281,12 +281,20 @@ func (cs *compileState) parseExpr(block *block, expr ast.Expr, markLocalVariable
 		if !ok {
 			return nil, nil, nil, false
 		}
+
+		if len(ss) != 0 {
+			stmts = append(stmts, ss...)
+		}
+
+		if len(es) == 0 {
+			return nil, nil, stmts, true
+		}
```

Comment:
Why do you need this fix?

Reply:
Because discard doesn't return a function, but a statement.
It avoids throwing this error
```go
if len(es) != 1 {
	cs.addError(e.Pos(), fmt.Sprintf("multiple-value context is not available at a callee: %s", e.Fun))
	return nil, nil, nil, false
}
```